src/bus/driver.c

```diff
@@ -348,10 +348,22 @@ static int driver_monitor(Bus *bus, Peer *sender, Message *message) {
 
                 r = connection_queue(&receiver->connection, NULL, message);
                 if (r) {
-                        if (r == CONNECTION_E_QUOTA)
+                        if (r == CONNECTION_E_QUOTA) {
+                                NameSet sender_names = NAME_SET_INIT_FROM_OWNER(&sender->owned_names);
```

Comment:
`sender` can be NULL here, right?

Reply:
Indeed, and moreover, the receiver cannot own any names. Fixed.